### ISSUE TITLE
Contribution Guideline and File Format Description

### ISSUE THREAD

**Issue by MartinWenge** [LeMonADE-project/LeMonADE#92]
We should add a contribution guideline including our programming style, for instance member functions start with lower case letter followed by camel case style for long function names. Suggestions of important points should be collected here in the discussion.

**Comment by MartinWenge** [comment]
We should also provide a description of the BFM file format in the [README](https://github.com/LeMonADE-project/LeMonADE/blob/master/README.md)

**Comment by tonimueller** [comment]
Concerning the output of some Features/MonomerExtensions I would suggest to define the standard to only write out the non-trivial behavior. Lets consider the FeatureConnectionSc. The standard is that only a few monomers in the system are reactive, but we write out this property for each monomer. For my opinion it would be sufficient to only write out the reactive ones. The Read-routine  for this property sets all monomers which do not appear in the bfm file to the standard value (nonreactive).
Would do you think about this?

**Comment by MartinWenge** [comment]
That's fine. The [FeatureSpringPotentialTwoGroups](include/LeMonADE/feature/FeatureSpringPotentialTwoGroups.h) is already working like this. Does that mean we have to rewrite Feature Attributes again?

**Comment by tonimueller** [comment]
Yes, if we decide to handle it like I suggest it,it would mean we have to revise FeatureAttributes and FeatureConnectionSc. Other Features are not affected, because they do not add any monomerExtension.

**Comment by MartinWenge** [comment]
note: typo in README in the section Contribution (programm -> program)

**Comment by MartinWenge** [comment]
note: the contribution from [Vinzenz Sinapius](https://github.com/vinzBad) should be added to the Authors.md file

**Comment by MartinWenge** [comment]
note: add publication
[Dendrimers in solution of linear polymers](https://pubs.acs.org/doi/10.1021/acs.macromol.9b00010)
some more publications made with LeMonADE?

**Comment by MartinWenge** [comment]
note: add publication
[Tendomers – force sensitive bis-rotaxanes with jump-like deformation behavior](https://pubs.rsc.org/en/content/articlelanding/2019/sm/c9sm00292h#!divAbstract)
